### PR TITLE
fix: Require qubit indices to be integers

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/circuits/test_qubit.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_index_lt_zero():
 
 @pytest.mark.xfail(raises=TypeError)
```
Comment (ajberdy) [inline test/unit_tests/braket/circuits/test_qubit.py:29]:
```suggestion
@pytest.mark.parametrize("qubit_arg", ("not a number", 0.5))
@pytest.mark.xfail(raises=TypeError)
```

File: test/unit_tests/braket/circuits/test_qubit.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def test_index_lt_zero():
...
 def test_index_non_int():
```
Comment (ajberdy) [inline test/unit_tests/braket/circuits/test_qubit.py:30]:
```suggestion
def test_index_non_int(qubit_arg):
```

File: test/unit_tests/braket/circuits/test_qubit.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def test_index_lt_zero():
...
-    Qubit("not a number")
+    Qubit(0.5)
```
Comment (ajberdy) [inline test/unit_tests/braket/circuits/test_qubit.py:31]:
```suggestion
    Qubit(qubit_arg)
```

Comment (ajberdy) [reply test/unit_tests/braket/circuits/test_qubit.py:31]: Sorry, wasn't sure how to do a multi-line suggestion lol

File: test/unit_tests/braket/circuits/test_qubit.py (hunk 1 above)
Comment (speller26) [reply test/unit_tests/braket/circuits/test_qubit.py:29]: Hah, juuuust in case

File: test/unit_tests/braket/circuits/test_qubit.py (hunk 3 above)
Comment (speller26) [reply test/unit_tests/braket/circuits/test_qubit.py]: You can drag down, just as you would for a comment,

File: src/braket/circuits/qubit.py [hunk 4]
```diff
@@ -36,8 +36,10 @@ def __new__(cls, index: int):
             >>> Qubit(0)
             >>> Qubit(1)
         """
+        if not isinstance(index, int):
```
Comment (kshitijc) [inline src/braket/circuits/qubit.py:39]: What's our stance on if someone supplies say numpy.int ? Should we accept that and cast it to int ?

**Review by kshitijc — APPROVED** [review]

**Pull request merged**